src/TS_HW3/Depаrtment.ts

```diff
@@ -0,0 +1,79 @@
+/*
+    Сутність Департамент - має назву, доменну область, список своїх співробітників і бюджет, що складається з дебіту і кредиту.
+
+    Так само у неї існують методи для обчислення балансу виходячи з поточного бюджету,
+    додавання нових співробітників, який враховує зміни балансу і перетворення з Попередньо найнятого на Співробітника або видалення Співробітника з минулого відділу.
+*/
+class Department {
+  private _staff: Employee[] = [];
+  private _budget: { debit: number; credit: number } = { debit: 0, credit: 0 };
+
+  get Staff(): Employee[] {
+    return this._staff;
+  }
+
+  get Budget(): { debit: number; credit: number } {
+    return this._budget;
+  }
+
+  constructor(
+    protected _company: Company,
+    public Name: DepartmentsNames,
+    public Domain: string,
+    debit: number,
+    credit: number
+  ) {
+    this._budget.debit = debit;
+    this._budget.credit = credit;
+    _company.Departments.push(this);
```

Comment:
It's right and what is important is it works. However, it concerns me a bit. What if you are going to refactor that class? What if you will add some additional fields and methods? Are we sure that we need everything in that class to push it into the company department's data structure? What I'm trying to explain is that that solution is not flexible and scalable from some perspective. I mean it's not optimal. It's not obvious what will be exactly the context in that moment.

Reply:
Do you mean that changes in Department class will be applied when Company will be already have some "old" items in it structure (DB)?

Reply:
Not really. It's more about data that we want to carry in the company's department data structure. I assume that we want to hold some information about departments, but not whole entities. It's just thoughts. It's not a suggestion to refactor something.